react/Gallery.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import GalleryLayout, { GalleryLayoutPropsWithSlots } from './GalleryLayout'
+//@ts-ignore
+import GalleryLegacy from './GalleryLegacy'
+
+export type MobileLayoutMode = 'normal' | 'small' | 'inline'
+
+export interface GalleryLegacyProps {
+  lazyItemsRemaining: number
+  maxItemsPerRow: number
+  minItemWidth: number
+  mobileLayoutMode: MobileLayoutMode
+  products: any[]
```

Comment:
Can't we type this based on the GraphQL?

Reply:
I typed it based on the shape (propType) received by the old gallery component